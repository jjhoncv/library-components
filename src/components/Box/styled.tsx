import styled from 'styled-components';

export const Box = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 34px 34px 0;
  width: 100%;
  position: relative;
  z-index: 1;
`;
