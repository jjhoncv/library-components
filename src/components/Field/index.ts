export { FieldGroup } from './styled';
export { Field } from './Field';
