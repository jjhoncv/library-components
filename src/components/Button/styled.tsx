import styled, { keyframes } from 'styled-components';
import { COLORS, FONTS } from './../../ui'

interface Props {
  fullWidth?: boolean;
  disabled?: boolean;
  loading?: boolean;
  primary?: boolean;
}

const animationLoading = keyframes`
  to {transform: rotate(360deg);}
`;

export const ButtonLoading = styled.div`
  display: inline-block;
  box-sizing: border-box;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 2px solid ${COLORS.BLACK};
  border-top: 2px solid ${COLORS.LIGHT_GRAY};
  animation: ${animationLoading} .6s linear infinite;

  ${(props: Props) => props.primary && `
    border: 2px solid ${COLORS.WHITE};
    border-top: 2px solid ${COLORS.BLACK};
  `};
`;

export const ButtonStyled = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${(props: Props) => props.fullWidth ? '100%' : 'auto'};
  background-color: ${COLORS.LIGHT_GRAY};
  height: 44px;
  padding: 12px 25px;
  border: none;
  border: 1px solid ${COLORS.OPAQUE_GRAY};
  border-radius: 4px;
  color: ${COLORS.BLACK};
  font-size: 14px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  cursor: pointer;

  &:hover {
    background-color: ${COLORS.HOVER_GRAY};
  }

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    user-select: none;
    opacity: 0.3;
  `};

  ${(props: Props) => props.loading && `
    pointer-events: none;
    user-select: none;
  `};

  ${(props: Props) => props.primary && `
    background:${COLORS.BLACK};
    color: ${COLORS.WHITE};

    &:hover {
      background-color: ${COLORS.HOVER_BLACK};
    }
  `};
`;
