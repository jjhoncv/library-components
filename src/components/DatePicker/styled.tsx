import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

interface WeekDateProps {
  disabled?: boolean;
  selected?: boolean;
}

export const DatePickerWrapper = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  font-family: ${FONTS.ROBOTO_REGULAR};

  ${(props: Props) => props.disabled && `
    pointer-events: none;

    ${DatePickerBox} {
      opacity: 0.3;
    }
  `}
`;

export const DatePickerIcon = styled.div`
  display: flex;
  flex-wrap: wrap;
  position: absolute;
  right: 8px;
  top: 8px;
  cursor: pointer;
  z-index: 1;

  ${(props: Props) => props.error && `
    svg {
      color: black;
    }
  `}
`;

export const DatePickerBox = styled.div`
  display: inline-flex;
  flex-wrap: wrap;
  align-items: center;
  width: 100%;
  position: relative;
  z-index: 0;
  color: ${COLORS.BLACK};
  font-size: 14px;
`;

export const DatePickerInput = styled.input`
  background-color: transparent;
  width: 100%;
  height: 40px;
  padding: 12px 30px 12px 12px;
  border: 2px solid ${COLORS.BORDER_GRAY};
  box-sizing: border-box;
  border-radius: 4px;
  cursor: pointer;
  z-index: 2;
  &::placeholder {
    font-family: ${FONTS.ROBOTO_REGULAR};
  }
  &:focus {
    outline: none;
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  &:focus ~ ${DatePickerIcon} {
    svg {
      color: black;
    }
  }
`;

export const DatePickerCalendarWrapper = styled.div`
  position: absolute;
  margin-top: 40px;
  z-index: 9;
  width: 240px;
  max-height: 300px;
  background-color: ${COLORS.WHITE};
  color: ${COLORS.BLACK};
  border: 1px solid ${COLORS.BORDER_GRAY};
  box-shadow: 0px 0px 8px ${COLORS.HOVER_GRAY};
  border-radius: 4px;
  padding: 5px;
`;

export const DatePickerMonth = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: bold;
`;

export const DatePickerPrev = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerNext = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;

export const DatePickerTitle = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 5 1 0;
  justify-content: center;
`;

export const DatePickerWeekDays = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
`;

export const DatePickerWeekDay = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  font-weight: bold;
`;

export const DatePickerDates = styled.div``;

export const DatePickerWeek = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

export const DatePickerWeekDate = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 0;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: all 0.3s linear;

  ${(props: WeekDateProps) => props.disabled && `
    color: ${COLORS.TEXT_GRAY} !important;
    cursor: not-allowed;

    &: hover {
      background: ${COLORS.WHITE} !important;
      color: ${COLORS.TEXT_GRAY} !important;
    }
  `}

  ${(props: WeekDateProps) => props.selected && `
    background-color: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  `}

  &:hover {
    background: ${COLORS.AQUA_BLUE};
    color: ${COLORS.WHITE};
  }
`;


