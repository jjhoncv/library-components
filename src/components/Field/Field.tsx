import * as React from 'react';
import * as PropTypes from 'prop-types';
import { FieldStyled, Label, BoxMessage } from './styled';

interface Props {
  className?: string;
  label?: string;
  portionWidth?: number;
  children: React.ReactNode;
  errorMessage?: string;
}

export const Field: React.FunctionComponent<Props> = ({
  className,
  label,
  portionWidth,
  children,
  errorMessage,
}) => {

  return (
    <FieldStyled className={className} portionWidth={portionWidth}>
      <Label>{label}</Label>
      {children}
      <BoxMessage errorMessage={errorMessage}>
        { errorMessage }
      </BoxMessage>
    </FieldStyled>
  );
};

Field.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  portionWidth: PropTypes.number,
  children: PropTypes.node.isRequired,
  errorMessage: PropTypes.string,
}
