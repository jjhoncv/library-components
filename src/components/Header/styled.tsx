import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

export const HeaderWrapper = styled.div`
  width: 100%;
`;
export const HeaderTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_BOLD};
  font-size: 24px;
  line-height: 30px;
  padding-bottom: 7px;
  border-bottom: 1px black dashed;
  color: ${COLORS.TEXT_GRAY};
`;
export const HeaderSubTitle = styled.span`
  display: block;
  font-family: ${FONTS.MULI_REGULAR};
  font-size: 18px;
  line-height: 23px;
  color: ${COLORS.BLACK};
  padding-top: 9px;
`;

export const HeaderTitleImage = styled(HeaderTitle)`
  display: flex;
  ${({ pos }) => (pos === "start") && `
    justify-content: flex-end;
    flex-direction: row-reverse;
    align-items: flex-end;
  `}

  ${({ pos }) => (pos === "end") && `
    align-items: flex-end;
    justify-content: space-between;
  `}
`;
