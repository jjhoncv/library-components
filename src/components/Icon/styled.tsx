import styled from 'styled-components';
import { COLORS } from './../../ui';

interface Props {
  color?: string,
  size?: string,
  colorHover?: string
}

export const IconStyled = styled.div`
  display: inline-flex;
  flex-wrap: wrap;

  svg {
    color: ${(props: Props) => props.color || `${COLORS.TEXT_GRAY}`}
    ${(props: Props) => `
      width: ${props.size || 'auto'};
      height: ${props.size || 'auto'};
    `}
  }

  &:hover {
    svg {
      color: ${(props: Props) => props.colorHover && `${props.colorHover}`};
    }
  }
`;
