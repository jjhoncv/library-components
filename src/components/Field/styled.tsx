import styled from 'styled-components';
import { SelectWrapper } from '../Select/styled';
import { DatePickerWrapper } from '../DatePicker/styled';
import { TimePickerWrapper } from '../TimePicker/styled';
import { TextAreaStyled } from '../TextArea/styled';

import { COLORS, FONTS } from '../../ui';

interface GroupProps {
  keepProps?: boolean;
}

interface Props {
  errorMessage?: string;
  portionWidth?: number;
}

export const FieldStyled = styled.div`
  font-family: ${FONTS.ROBOTO_REGULAR};
  width: ${props => `${props.portionWidth * 100}%`};
  flex: 1 1 100%;


  input {
    width: 100%;
  }
  ${SelectWrapper} {
    width: 100%;
  }
  ${DatePickerWrapper} {
    width: 100%;
  }
  ${TimePickerWrapper} {
    width: 100%;
  }
  ${TextAreaStyled} {
    width: 100%;
  }
  margin-bottom: 14px;

  @media(min-width: 480px) {
    ${(props: Props) => props.portionWidth && `
      flex: 1 1 calc(${props.portionWidth * 100}% - 20px);
    `}
  }
`;

export const FieldGroup = styled.div`
  display: flex;
  flex-wrap: wrap;

  ${(props: GroupProps) => props.keepProps && `
    ${FieldStyled} {
      margin-right: 20px;
      flex: 1 1 calc(50% - 20px);
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  `}

  @media(min-width: 480px) {
    ${FieldStyled} {
      margin-right: 20px;
    }
    ${FieldStyled}:last-child {
      margin-right: 0;
    }
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 2px;
  min-height: 19px;
  color: ${COLORS.BLACK};
  font-size: 12px;

  @media(min-width: 480px) {
    font-size: 14px;
  }
`;

export const BoxMessage = styled.span`
  display: block;
  font-size: 12px;
  line-height: 15px;
  margin-top: 4px;
  ${(props: Props) => props.errorMessage && `
    color: ${COLORS.ERROR_RED};
  `}
  height: 15px;
`;
