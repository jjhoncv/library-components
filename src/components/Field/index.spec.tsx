import * as React from 'react';
import { shallow } from 'enzyme';
import { Field } from './Field';
import { Input } from './../Input';
import { BoxMessage } from './styled';

let propsField = {
  className: 'c-field',
  label: 'Nombre'
}

describe('Field Component', () => {
  describe('Field => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <Field {...propsField}>
          <Input name='input'/>
        </Field>
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('Field Builder', () => {
    it('Field => validate label', () => {
      const wrapper = shallow(
        <Field {...propsField}>
          <Input name='input'/>
        </Field>
      );
      expect(wrapper.childAt(0).prop('children')).toBe('Nombre');
    })
    it('Field => render with prop portionWidth', () => {
      const props = {
        className: 'c-field',
        label: 'Nombre',
        portionWidth: 1/4
      }
      const wrapper = shallow(
        <Field {...props}>
          <Input name='input'/>
        </Field>);
      expect(wrapper.prop('portionWidth')).toEqual(1/4);
    })
    it('Field => validate errorMessage', () => {
      const props = {
        className: 'c-field',
        label: 'Nombre',
        portionWidth: 1/4,
        errorMessage: 'Campo Requerido'
      }
      const wrapper = shallow(
        <Field {...props}>
          <Input name='input'/>
        </Field>);
      expect(wrapper.find(BoxMessage).prop('errorMessage')).toBe('Campo Requerido');
    })
  });
});


