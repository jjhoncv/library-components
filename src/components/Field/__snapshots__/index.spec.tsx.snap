// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Component Field => render snapshot Render correctly 1`] = `"<div class=\\"c-field sc-ckVGcZ jQapgt\\"><label class=\\"sc-eNQAEJ hnchOF\\">Nombre</label><div class=\\"sc-kEYyzF ipzqjR\\"><input type=\\"text\\" class=\\"sc-kkGfuU kUpoXX\\" name=\\"input\\" width=\\"200px\\"/></div><span class=\\"sc-hMqMXs kEljjE\\"></span></div>"`;
